### PR TITLE
ARCO-205: use cache for updating txs in metamorph

### PR DESCRIPTION
**Pull request by pawellewandowski98**
## Description of Changes

Use cache in metamorph to update tx

## Linked Issues / Tickets

ARCO-205

## Testing Procedure

Describe the tests you've added or any testing steps you've taken.

- [ ] I have added new unit tests
- [ ] All tests pass locally
- [ ] I have tested manually in my local environment

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have updated `CHANGELOG.md` with my changes

### CODE REVIEW

File: internal/metamorph/store/store.go [hunk 1]
```diff
@@ -83,8 +84,52 @@ type MetamorphStore interface {
...
+	Status       metamorph_api.Status `json:"status"`
+	Error        error                `json:"-"`
+	CompetingTxs []string             `json:"competing_txs"`
+	// Fields for marshalling
+	HashStr  string `json:"hash"`
+	ErrorStr string `json:"error"`
+}
+
+// UnmarshalJSON Custom method to unmarshall the UpdateStatus struct
+func (u *UpdateStatus) UnmarshalJSON(data []byte) error {
```
Comment (arkadiuszos4chain) [inline internal/metamorph/store/store.go:97]: Do you use them?

File: internal/metamorph/processor.go [hunk 2]
```diff
@@ -397,28 +400,41 @@ func (p *Processor) StartProcessStatusUpdatesInStorage() {
 	go func() {
 		defer p.waitGroup.Done()
 
-		statusUpdatesMap := map[chainhash.Hash]store.UpdateStatus{}
+		//statusUpdatesMap := map[chainhash.Hash]store.UpdateStatus{}
```
Comment (arkadiuszos4chain) [inline internal/metamorph/processor.go:403]: remove

File: internal/metamorph/processor.go [hunk 3]
```diff
@@ -397,28 +400,41 @@ func (p *Processor) StartProcessStatusUpdatesInStorage() {
...
+					return
+				}
 
-				if len(statusUpdatesMap) >= p.processStatusUpdatesBatchSize {
-					p.checkAndUpdate(statusUpdatesMap)
-					statusUpdatesMap = map[chainhash.Hash]store.UpdateStatus{}
+				if len(actualUpdateStatusMap) >= p.processStatusUpdatesBatchSize {
+					p.checkAndUpdate(actualUpdateStatusMap)
+
+					err = p.cacheStore.Del(CacheStatusUpdateKey)
```
Comment (arkadiuszos4chain) [inline internal/metamorph/processor.go:420]: Shouldn't be a part of `checkAndUpdate(...)`?

File: internal/metamorph/processor.go (hunk 2 above)
Comment (pawellewandowski98) [reply internal/metamorph/processor.go]: Removed

File: internal/metamorph/processor.go (hunk 3 above)
Comment (pawellewandowski98) [reply internal/metamorph/processor.go]: Moved

**Review by kuba-4chain — APPROVED** [review]

**Review by arkadiuszos4chain — APPROVED** [review]

**Pull request merged**